### PR TITLE
Split swift-toolchain.yml triggers into seperate workflows

### PR DESCRIPTION
**Pull request by kendalharland**
This PR refactors `swift-toolchain.yml` so that it has no automated triggers. Instead it is only run via one of:
- `workflow_call` 
- `workflow_dispatch`. 

This PR also adds three new workflows that each include a `vars.USE_CIRUN` kill switch:
- **pull-request-swift-toolchain-github.yml**
  - runs `swift-toolchain.yml` with GitHub images on `pull_request`
  - does not create releases
- **pull-request-swift-toolchain-cirun.yml**
  - runs `swift-toolchain.yml` with Cirun images on `pull_request`
  - does not create releases 
- **schedule-swift-toolchain.yml**
  - runs `swift-toolchain.yml` every 6 hours with Azure images on `schedule`
  - creates releases

Separate workflows must be used for the pull-request* workflows instead of one parent workflow that uses GitHub and Cirun images via a `matrix`. This is because the child workflows called from the parent would upload artifacts with identical names, and race. The first uploader wins while the second fails. It is easier to handle this using separate workflow files than it is to - for example - namespace the uploads using a string prefix, inside swift-toolchain.yml

swft-toolchain.yml is also updated with these inputs:
- `default_runner`
  - The runner to use for nearly all steps. "default_runner" is used instead of "windows_runner" to make this behavior more obvious.
-  `compiler_runner`
   - The runner to use for compiler steps
- `publish_artifacts`
  - controls whether artifacts are uploaded.

### CODE REVIEW

**Comment by kendalharland** [comment]
Happy to upstream the changes to the workflow inputs to compnerd/swift-build. Still sending this for review in the meantime so I can feedback while the tests run.

File: .github/workflows/swift-toolchain.yml [hunk 1]
```diff
@@ -27,6 +25,24 @@ on:
...
+        default: 'windows-latest'
+        required: false
+        type: string
+
+      compilers_runner:
+        description: 'The runner to use for building the compilers in this worflow. If unset, uses default_runner'
+        type: string
+        required: false
+
+      dry_run:
```
Comment (tristanlabelle) [inline .github/workflows/swift-toolchain.yml:40]: `publish_artifacts` would be clearer a name

File: .github/workflows/swift-toolchain.yml [hunk 2]
```diff
@@ -27,6 +25,24 @@ on:
         description: 'Code Sign'
         default: false
         type: boolean
+
+      default_runner:
```
Comment (tristanlabelle) [inline .github/workflows/swift-toolchain.yml:29]: Can we move these changes to upstream?

**Review by tristanlabelle — APPROVED** [review]
Great idea!

File: .github/workflows/swift-toolchain.yml (hunk 1 above)
Comment (kendalharland) [reply .github/workflows/swift-toolchain.yml:40]: Done in https://github.com/compnerd/swift-build/pull/747. I'll delete and rebase once that's merged

File: .github/workflows/swift-toolchain.yml (hunk 2 above)
Comment (kendalharland) [reply .github/workflows/swift-toolchain.yml:29]: Done: https://github.com/compnerd/swift-build/pull/747

**Comment by kendalharland** [comment]
Upstreamed most of the changes to swift-toolchain.yml and will rebase once https://github.com/compnerd/swift-build/pull/747 is merged.

**Comment by kendalharland** [comment]
This PR is failing to package swift-inspect on cirun because of an if-statement. This will be fixed by https://github.com/compnerd/swift-build/pull/749

**Comment by kendalharland** [comment]
Update, this change is g2g, just waiting on the base swift-toolchain.yml build to turn green. @z2oh is currently working to resolve.

**Pull request merged**